src/pip/_internal/legacy_resolve.py

```diff
@@ -267,6 +267,10 @@ def _check_skip_installed(self, req_to_install):
         # requirements we have to pull the tree down and inspect to assess
         # the version #, so it's handled way down.
         if not req_to_install.link:
+            if req_to_install.is_pinned:
```

Comment:
Should we be checking that the pinned does indeed match the installed version here?

I imagine this'll return even if we're at `0.1.0` if pinned on `0.1.1` for example.

Reply:
We already know in this path that `req_to_install.satisfied_by` is truthy, so if the version must match.